### PR TITLE
Docs for parallel execution of CI

### PR DESCRIPTION
**Pull request by enzofab91**
#### Description:
Generating documentation for executing CI in parallel.

### CODE REVIEW

File: docs/ci.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# CI
+
+## Parallelization with Parallel & Knapsack
+Knapsack and Parallel gems allow us to run tests in several nodes at the same time, benefiint us in the execution time. Knapsack it parallelizes at node level while Parallel does it at CPU level.
```
Comment (lauperalti) [inline docs/ci.md:4]: benefiting?

File: docs/ci.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+To schedule the cron task you have to do it in `.github/workflows/update_knapsack_report.yml:6`
+Now is setted on February 31th so will never run. 
+
+```sh
+  - cron: '0 5 31 2 *'    
+  # The above cron does not run. Replace with the wanted periodicity.
+```
+## Coverage
+When splitting tests in different nodes each one reports only coverate of a part of the code files are testing.
```
Comment (lauperalti) [inline docs/ci.md:39]: This is a bit confusing for me. maybe rephrase it to: When splitting tests in different nodes, each report covers only a part of the code files being tested. WDYT?

**Review by lauperalti — APPROVED** [review]
💪

File: docs/ci.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+On Github Actions you can add any nodes you want using matrix strategy, setting up some variables:
+
+```sh
+  ci_node_total: [4]
+  # set N-1 indexes for parallel jobs
+  # When you run 2 parallel jobs then first job will have index 0, the second job will have index 1 etc
+  ci_node_index: [0, 1, 2, 3]
+```
+
+To update cpu cores quantity on every onde you can do it by updating this variable:
```
Comment (mrodriguez90) [inline docs/ci.md:20]:
```suggestion
To update cpu cores quantity on every node you can do it by updating this variable:
```

File: docs/ci.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+To update tests on local machine you can by executing `KNAPSACK_CI_NODE_TOTAL=4 KNAPSACK_CI_NODE_INDEX=1 PARALLEL_TESTS_CONCURRENCY=2 bundle exec parallel_rspec -n 2 -e './bin/parallel_tests'`. This will run subset of tests files corresponding to second node.
+
+## Generating report
+Knapsack report needs to be updated frequently to balance execution time among nodes. This can be done manually by executing:
+`knapsack_generate_generate_report: true bundle exec knapsack:rspec`
+
+It is also recommended to generate the report in the CI for a better precision. For this you have available a workflow in Github Actions that triggers the report generation and creates a pull request automatically. This workflow can be scheduled in the frequency you want or even can be manually triggered.
+
+To schedule the cron task you have to do it in `.github/workflows/update_knapsack_report.yml:6`
+Now is setted on February 31th so will never run.
```
Comment (mrodriguez90) [inline docs/ci.md:32]: `It is now scheduled for February 31 so will never run. `

File: docs/ci.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+To schedule the cron task you have to do it in `.github/workflows/update_knapsack_report.yml:6`
+Now is setted on February 31th so will never run. 
+
+```sh
+  - cron: '0 5 31 2 *'    
+  # The above cron does not run. Replace with the wanted periodicity.
+```
+## Coverage
+When splitting tests in different nodes, each report covers only a part of the code files being tested.
+For this reason a job in the CI is added to sums coverages from all nodes to be uses by SimpleCov. This job will be executed after all nodes have finished and will send the final report to CodeClimate.
```
Comment (mrodriguez90) [inline docs/ci.md:40]: `For this reason a job in the CI is added to sums coverages from all nodes to be used by SimpleCov.`

File: docs/ci.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# CI
+
+## Parallelization with Parallel & Knapsack
+Knapsack and Parallel gems allow us to run tests in several nodes at the same time, benefiting us in the execution time. Knapsack it parallelizes at node level while Parallel does it at CPU level.
```
Comment (grosendo2006) [inline docs/ci.md:4]:
```suggestion
Knapsack and Parallel gems allow us to run tests in several nodes at the same time, benefiting us in the execution time. Knapsack parallelizes it at node level while Parallel does it at CPU level.
```

**Review by grosendo2006 — APPROVED** [review]

**Review by mrodriguez90 — APPROVED** [review]

**Review by jpascual1994 — APPROVED** [review]

**Review by rodrieiz — APPROVED** [review]
🤩

File: README.md [hunk 7]
```diff
@@ -105,6 +107,7 @@ To illustrate, `bin/rails console` will run the console in the docker container
 - Set your [frontend URL](https://github.com/cyu/rack-cors#origin) in `config/initializers/rack_cors.rb`
 - Set your mail sender in `config/initializers/devise.rb`
 - Config your timezone accordingly in `application.rb`.
+- Config CI in parallel execution. See [docs](docs/ci.md).
```
Comment (santib) [inline README.md:110]: 👏

File: docs/ci.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  ci_node_total: [4]
+  # set N-1 indexes for parallel jobs
+  # When you run 2 parallel jobs then first job will have index 0, the second job will have index 1 etc
+  ci_node_index: [0, 1, 2, 3]
+```
+
+CPU cores quantity on every node are obtained automatically from Github Actions config `echo "cpu_cores=$(nproc)" >> $GITHUB_ENV`.
+
+If you want to update it manually you can do it by updating this variable:
+`PARALLEL_TESTS_CONCURRENCY: 2`
```
Comment (santib) [inline docs/ci.md:23]: I don't think someone would want to do this 🤔

File: docs/ci.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+It is also recommended to generate the report in the CI for a better precision. For this you have available a workflow in Github Actions that triggers the report generation and creates a pull request automatically. This workflow can be scheduled in the frequency you want or even can be manually triggered.
+
+To schedule the cron task you have to do it in `.github/workflows/update_knapsack_report.yml:6`
+It is now scheduled for February 31 so will never run.
+
+```sh
+  - cron: '0 5 31 2 *'    
+  # The above cron does not run. Replace with the wanted periodicity.
+```
```
Comment (santib) [inline docs/ci.md:39]:
```suggestion
```

```

File: Gemfile [hunk 10]
```diff
@@ -8,7 +8,7 @@ gem 'rails', '~> 7.0.7'
 gem 'activeadmin', '~> 2.9'
 gem 'active_storage_base64', '~> 2.0.0'
 gem 'arctic_admin', '~> 3.3.0'
-gem 'aws-sdk-s3', '~> 1.132', require: false
+gem 'aws-sdk-s3', '~> 1.134', require: false
```
Comment (santib) [inline Gemfile:11]: ??

File: README.md [hunk 11]
```diff
@@ -81,9 +81,11 @@ To illustrate, `bin/rails console` will run the console in the docker container
 - [Faker](https://github.com/stympy/faker) for generating test data
 - [Flipper](https://github.com/jnunemaker/flipper) for feature flag support
 - [Jbuilder](https://github.com/rails/jbuilder) for json views
+- [Knapsack](https://github.com/KnapsackPro/knapsack) for generating CI time report and splitting into nodes
 - [Letter Opener](https://github.com/ryanb/letter_opener) for previewing a mail in the browser
 - [Oj](https://github.com/ohler55/oj) for optimized json
 - [Pagy](https://github.com/ddnexus/pagy) for pagination
+- [Parallel](https://github.com/grosser/parallel_tests) for split CI in several cores
```
Comment (santib) [inline README.md:88]:
```suggestion
- [Parallel Tests](https://github.com/grosser/parallel_tests) for split CI in several cores
```

File: docs/ci.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+CPU cores quantity on every node are obtained automatically from Github Actions config `echo "cpu_cores=$(nproc)" >> $GITHUB_ENV`.
+
+If you want to update it manually you can do it by updating this variable:
+`PARALLEL_TESTS_CONCURRENCY: 2`
+
+To update tests on local machine you can by executing `KNAPSACK_CI_NODE_TOTAL=4 KNAPSACK_CI_NODE_INDEX=1 PARALLEL_TESTS_CONCURRENCY=2 bundle exec parallel_rspec -n 2 -e './bin/parallel_tests'`. This will run subset of tests files corresponding to second node.
+
+## Generating report
+Knapsack report needs to be updated frequently to balance execution time among nodes. This can be done manually by executing:
+`KNAPSACK_GENERATE_REPORT=true bundle exec rspec spec: true bundle exec knapsack:rspec`
```
Comment (santib) [inline docs/ci.md:28]:
```suggestion
`KNAPSACK_GENERATE_REPORT=true bundle exec knapsack:rspec`
```
?

File: docs/ci.md (hunk 8 above)
Comment (rodrieiz) [reply docs/ci.md:22]: I'm not sure whether to remove it; it might be useful if the machine has many cores to limit parallelism or to deactivate it temporarily, wdyt?

**Review by santib — APPROVED** [review]
